engine/src/main/resources/assets/shaders/ssao_frag.glsl

```diff
@@ -69,6 +69,10 @@ void main() {
         sampleDepth = reconstructViewPos(texture2D(texDepth, offset.xy).r * 2.0 - 1.0, gl_TexCoord[0].xy, invProjMatrix).z;
 
         rangeCheck = smoothstep(0.0, 1.0, ssaoRad / abs(viewSpacePos.z - sampleDepth));
+        const float maxDepthDifference = 2;
+        if (abs(viewSpacePos.z - sampleDepth) > maxDepthDifference) {
+        	rangeCheck = 0;
+        }
```

Comment:
Looks much better, thank you. Stand-by for some testing on the value of maxDepthDifference.